internal/pipeline/sign_test.go

```diff
@@ -22,12 +21,7 @@ func TestSignVerify(t *testing.T) {
 			},
 			{
 				Source: "another-plugin#v3.4.5",
-				Config: ordered.MapFromItems(
-					ordered.TupleSA{
-						Key:   "llama",
-						Value: "Kuzco",
-					},
-				),
+				Config: map[string]any{"llama": "Kuzco"},
```

Comment:
Given that this PR was made to address the order of keys in plugin config, we should have a test that would have failed if the order weren't stable.

It might be impossible to write such a test using the `json` marshaller in go, but it's still worth having. The fact that `json.Marshal` sorts the keys before writing them is pretty crucial, so we should assert in our tests that it does to prevent regression.

Reply:
I've added a test that I think should capture encoding stability problems. It failed, but the reason was how I was holding Sign and Verify, so I made their argument processing more consistent.